ksql-engine/src/main/java/io/confluent/ksql/util/QueryIdGeneratorUsingOffset.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.util;
+
+public class QueryIdGeneratorUsingOffset implements QueryIdGenerator {
```

Comment:
I don't think we need to couple the name to the use of `offsets`.  The class doesn't care where that the next id comes from the message offset. 

Consider:

```suggestion
public class SpecificQueryIdGenerator implements QueryIdGenerator {
```

Or something else that reflects that it's just providing the id it was set with.

Also, a javadoc explaining what this type does (and the others) wouldn't go amiss.